app/screens/post_options/post_options.tsx

```diff
@@ -63,17 +72,61 @@ const PostOptions = ({
     const shouldRenderFollow = !(sourceScreen !== Screens.CHANNEL || !thread);
     const shouldShowBindings = bindings.length > 0 && !isSystemPost;
 
-    const snapPoints = [
+    const snapPoints = useMemo(() => {
+        const items: BottomSheetProps['snapPoints'] = [1];
+        let count = shouldShowBindings ? 0.5 : 0;
+        if (canReply && sourceScreen !== Screens.THREAD) {
+            count++;
+        }
+
+        if (shouldRenderFollow) {
+            count++;
+        }
+
+        if (canMarkAsUnread && !isSystemPost) {
+            count++;
+        }
+
+        if (canCopyPermalink) {
+            count++;
+        }
+
+        if (!isSystemPost) {
+            count++;
+        }
+
+        if (canCopyText && post.message) {
+            count++;
+        }
+
+        if (canPin) {
+            count++;
+        }
+
+        if (canEdit) {
+            count++;
+        }
+
+        if (canDelete) {
+            count++;
+        }
+
+        items.push(bottomSheetSnapPoint(count, ITEM_HEIGHT, bottom) + (canAddReaction ? REACTION_PICKER_HEIGHT + REACTION_PICKER_MARGIN : 0));
+
+        if (shouldShowBindings) {
+            items.push('90%');
+        }
+
+        return items;
+    }, [
```

Comment:
```suggestion
    
    const countSnapPoints = (conditions: boolean[]) => {
        return conditions.filter(Boolean).length;
    };

    const snapPoints = useMemo(() => {
        const items: BottomSheetProps['snapPoints'] = [1];
        let count = shouldShowBindings ? 0.5 : 0;
        count += countSnapPoints([
            canReply && sourceScreen !== Screens.THREAD,
            shouldRenderFollow,
            canMarkAsUnread && !isSystemPost,
            canCopyPermalink,
            !isSystemPost,
            Boolean(canCopyText && post.message),
            canPin,
            canEdit,
            canDelete,
        ]);

        items.push(bottomSheetSnapPoint(count, ITEM_HEIGHT, bottom) + (canAddReaction ? REACTION_PICKER_HEIGHT + REACTION_PICKER_MARGIN : 0));

        if (shouldShowBindings) {
            items.push('90%');
        }

        return items;
    }, [
```

Reply:
Could do something like this if all the `if` statements look ugly and make the conditions more compact to view line by line. 0/5

Reply:
Thanks for the suggestion, I went with a different approach.

```ts
const optionsCount = [
    canCopyPermalink, canCopyText, canDelete, canEdit,
    canMarkAsUnread, canPin, canReply, !isSystemPost, shouldRenderFollow,
].reduce((acc, v) => {
    return v ? acc + 1 : acc;
}, 0) + (shouldShowBindings ? 0.5 : 0);
```